subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskParametersIntegrationTest.groovy

```diff
@@ -778,6 +778,37 @@ task someTask(type: SomeTask) {
         "dirs"  | "output-file.txt"
     }
 
+    @ValidationTestFor(
+        ValidationProblemId.CANNOT_WRITE_OUTPUT
+    )
+    @Issue("https://github.com/gradle/gradle/issues/15679")
+    def "fileTrees with regular file roots cannot be used as output files"() {
```

Comment:
It would be good, somehow, if we could capture the information that the problem comes from having a file tree, so that we can provide a better error message in this case and suggest another solution, but I have no clear idea how to do this.

Reply:
I added a new validation for file trees only. Can you take another look?